src/game/SaveLoadGame.cc

```
@@ -1132,6 +1138,85 @@ static void SaveMercProfiles(HWFILE const f)
 	}
 }
 
+const char * IMPSavedProfileCreateFilename(const char *nickname_cstring)
+{
+	char *profile_filename = (char *)malloc(strlen(GCM->getSavedGamesFolder().c_str()) + 32);
+	sprintf(profile_filename, "%s/mercprofile.%s",GCM->getSavedGamesFolder().c_str(), nickname_cstring);
+	return profile_filename;
+}
+
+bool IMPSavedProfileDoesFileExist(const char *nickname)
+{
+	const char *profile_filename = IMPSavedProfileCreateFilename(nickname);
+	bool fexists = Fs_exists(profile_filename);
+	free((void*)profile_filename);
+	return fexists;
+}
+
+SGPFile* const IMPSavedProfileOpenFileForRead(const char *nickname)
+{
+	if (!IMPSavedProfileDoesFileExist(nickname)) return (SGPFile *)-1;
```

We can just throw an exception here.